src/systray.cc

```diff
@@ -0,0 +1,101 @@
+#include "systray.h"
+#include "channel.h"
+#include "sinkwidget.h"
+
+Systray::Systray(MainWindow &parent) :
+    mw(parent),
+    muted(false) {
+
+    systrayQuitAction.setText(tr("&Quit"));
+    systrayRestoreAction.setText(tr("&Restore"));
+    systrayMinimizeAction.setText(tr("Mi&nimize"));
+
+    connect(&systrayMinimizeAction, &QAction::triggered, &mw, &MainWindow::hide);
+    connect(&systrayRestoreAction, &QAction::triggered, &mw, &MainWindow::showNormal);
+    connect(&systrayQuitAction, &QAction::triggered, &mw, &MainWindow::quit);
+    connect(this, &QSystemTrayIcon::activated, this, &Systray::iconActivated);
+
+    systrayIconMenu.addAction(&systrayMinimizeAction);
+    systrayIconMenu.addAction(&systrayRestoreAction);
+    systrayIconMenu.addSeparator(); 
+    systrayIconMenu.addAction(&systrayQuitAction);
+
+    QSize sz(256,256);
+    systrayIcons[NOT_MUTED].addPixmap(mw.style()->standardIcon(QStyle::SP_MediaVolume).pixmap(sz));
+    systrayIcons[MUTED].addPixmap(mw.style()->standardIcon(QStyle::SP_MediaVolumeMuted).pixmap(sz));
+
+    setIcon(systrayIcons[NOT_MUTED]);
+    setContextMenu(&systrayIconMenu);
+    
+    qApp->installEventFilter(this);
```

Comment:
Why are you installing event filter for all events ever... you just need/want to get your own events. So you just need to override `void event(QEvent *e)` and not to check every one event of the whole application. Not?